File: Lab2/IS_Lab2_JSON/deserialize_json.py
# -*- coding: utf-8 -*-

"""
  deserialize json
"""

import json



class DeserializeJson:
    def __init__(self, filename):
        print("let's deserialize something")
        tempdata = open(filename, encoding="utf8")
        self.data = json.load(tempdata)

    # przykładowe statystyki
    def somestats(self):
        example_stat = 0
        for dep in self.data:
            if dep['typ_JST'] == 'GM' and dep['Województwo'] == 'dolnośląskie':
                example_stat += 1
        print('liczba urzędów miejskich w województwie dolnośląskim: ' + ' ' + str(example_stat))




        typy = set()
        for dep in self.data:
            typy.add(dep['typ_JST'])
        print(typy)
        wojewodztwa_nazwy = set()

        for dep in self.data:
            wojewodztwa_nazwy.add(dep['Województwo'].strip())
        print(wojewodztwa_nazwy)

        for wojewodztwo in wojewodztwa_nazwy:
            print('\n'+wojewodztwo)
            for typ in typy:
                count = 0
                for dep in self.data:
                    if dep['typ_JST'] == typ and dep['Województwo'] == wojewodztwo:
                        count += 1
                print(typ+': '+ str(count))

File: Lab2/C#/IS_Lab2_JSON/IS_Lab2_JSON/deserialize_json.cs
using Newtonsoft.Json;
using System.Collections.Specialized;
using System.Runtime.CompilerServices;

public class DeserializeJson
{
    public DeserializeJson(string filepath)
    {
        Console.WriteLine("let's deserialize something");
        var tempdata = new StreamReader(filepath,encoding:System.Text.Encoding.UTF8);
        var json = tempdata.ReadToEnd();
        Data = JsonConvert.DeserializeObject<List<Dictionary<string, object?>>>(json)!;
    }



    public List<Dictionary<string, object?>> Data { get; set; }

    public void somestats()
    {
        HashSet<String> typy = new HashSet<String>();
        HashSet<String> wojewodztwa_nazwy = new HashSet<String>();

        foreach (var dep in Data)
        {
            typy.Add(dep["typ_JST"]!.ToString()?.Trim());
            var woj = dep["Województwo"]!.ToString()?.Trim();
            if(woj != null) wojewodztwa_nazwy.Add(woj);
        }
        typy.ToList<String>().ForEach(x => Console.WriteLine(x));
        wojewodztwa_nazwy.ToList<String>().ForEach(x => Console.WriteLine(x));

        foreach( string wojewodztwo in wojewodztwa_nazwy)
        {
            Console.WriteLine(wojewodztwo);
            foreach(string typ in typy)
            {
                var count = 0;
                foreach(var dep in Data)
                {
                    if (dep["typ_JST"].ToString() == typ && dep["Województwo"].ToString() == wojewodztwo)
                    {
                        count += 1;
                    }
                }
                Console.WriteLine(typ + ": " + count.ToString());
            }
        }
    }
}